socket/server.go

```diff
@@ -70,11 +74,22 @@ func (s *Server) handleConn(conn net.Conn) {
 
 	for {
 		event, args, err := readWriter.Read()
+		// if there is an error occurred in read or write data we entered the if block
 		if err != nil {
-			if errors.Is(err, io.EOF) {
+			// this condition is not enough and like client-side code you should check all conditions
+
+			//if errors.Is(err, io.EOF) {
+			//break
+			//}
+
+			// if each one of these conditions set true we just break this loop
+			if isConnectionResetError(err) || strings.ContainsAny(err.Error(), "closed") ||
+				err == io.EOF || strings.ContainsAny(err.Error(), "reset") || strings.ContainsAny(err.Error(), "EOF") {
```

Comment:
Some concerns:
1. I think checking strings is an overkill. Is there a reason why they should be checked explicitly?
2. `err == io.EOF`, why not `errors.Is(err, io.EOF)`?